### PR TITLE
[MM-24453] Sanitize user term search

### PR DESCRIPTION
**Pull request by ethervoid**
#### Summary

Remove the @ at the beginning of the search term in order to
make it possible for ES to return results

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-24453

### CODE REVIEW

File: store/searchlayer/user_layer.go [hunk 1]
```diff
@@ -142,6 +145,10 @@ func (s *SearchUserStore) autocompleteUsersInChannelByEngine(engine searchengine
 	return autocomplete, nil
 }
 
+func (s *SearchUserStore) sanitizeTerm(term string) string {
+	return strings.TrimLeft(term, "@")
+}
```
Comment (mgdelacroix) [inline store/searchlayer/user_layer.go:150]: nit: if we're not using the store as part of this function, we can move it to an external file like `utils.go` in the same package to be an independent method

Comment (ethervoid) [reply store/searchlayer/user_layer.go:150]: Good point. On it

Comment (ethervoid) [reply store/searchlayer/user_layer.go]: Done

**Comment by ethervoid** [comment]
@isacikgoz I've been thinking about where to put this method and although we have sanitization in the app layer we need to include it in the store layer because the store methods can be used anywhere.

I think the search layer is a perfect for because it works for all the search engines and we don't have to duplicate it inside every engine code

**Pull request merged**